src/test/java/com/alem/todolist/ToDoListControllerTest.java

```diff
@@ -66,14 +63,15 @@ void shouldReturnFirstTask() throws Exception {
                         .accept("application/json"))
                         .andExpect(status().isOk());
     }
-
     @Test
-    void shouldReturnAllTasks() throws Exception  {
+    void shouldReturnAllTasks() throws Exception {
+
         List<Task> tasks = toDoListService.fetchAllTasks();
         when(toDoListRepository.findAll()).thenReturn(tasks);
-        this.mockmvc.perform(get("/tasks").accept("application/json"))
-                .andExpect(MockMvcResultMatchers.jsonPath("$", hasSize(7)))
-                .andExpect(status().isOk());
+        this.mockmvc.perform(MockMvcRequestBuilders.get("/tasks"))
+                .andExpect(MockMvcResultMatchers.jsonPath("$", hasSize(tasks.size())))
```

Comment:
You did not really test anything here. Tasks have a size of 0, so after mocked request, it will also be 0.
You have to mock what repository is returning, and then tests mock mvc request